#In the first code, x prints 3
#In the second code, an error occurs because there is no x in the outer scope.
